spec/lib/miq_expression_spec.rb

```diff
@@ -2245,6 +2275,38 @@
       )
     end
 
+    it "return column info for model.virtualassociation..virtualassociation-invalid" do
+      field = "ManageIQ::Providers::InfraManager::Vm.service.user.vms-invalid"
+      col_info = described_class.get_col_info(field)
+      expect(col_info).to match(
+        :data_type                      => nil,
+        :excluded_by_preprocess_options => false,
+        :format_sub_type                => nil,
+        :include                        => {},
+        :tag                            => false,
+        :virtual_column                 => false,
+        :sql_support                    => false,
+        :virtual_reflection             => true
+      )
+    end
+
+    it "return column info for model.invalid-active" do
+      field = "ManageIQ::Providers::InfraManager::Vm.invalid-active"
+#      expect { described_class.get_col_info(field) }.to raise(ArgumentError)
+      col_info = described_class.get_col_info(field)
+      expect(col_info).to match(
+        :data_type                      => nil,
+        :excluded_by_preprocess_options => false,
+      # :format_sub_type                => nil,
+        :include                        => {:invalid => {}}, # unsure - should be {}
+        :tag                            => false,
+        :virtual_column                 => true, # unsure - should be false?
+        :sql_support                    => false,
+        :virtual_reflection             => true
+      )
+    end
+
+    # don't think this is valid -- this is not sql friendly
```

Comment:
No issues. I will have to get into the U.K. And see how it behaves with invalid input